common/api/core-frontend.api.md

```diff
@@ -7734,9 +7737,9 @@ export class MapLayerFeatureRecord {
 
 // @public
 export class MapLayerFormat {
-    // @internal
+    // @beta

```

Comment:
`createImageryProvider` and `createMapLayerTree` should be `abstract`. The default implementations `assert` at run-time presumably to tell the developer that they were supposed to overrde these functions. Far better to express that at compile time.

Reply:
Part of promoting from internal to beta means refactoring APIs to make them more usable.
I'm not sure what your concern is about the methods being called statically. Yes, they are static methods. They still need to be overridden to do anything useful.

Reply:
I think you're telling me that `static` methods cannot also be `abstract`.
That is indeed a problem.

Reply:
Dunno why MapLayerFormat consists of only static methods, but can't change it now without breaking the public API.
Please add a `return undefined` to `createImageryProvider` and add a message to the two `asserts` to make it clear to the developer that they are supposed to override these methods.